### ISSUE TITLE
consistency loss on fcos

### ISSUE THREAD

**Issue by ZJUZQ** [soo89/CSD-SSD#4]
thank you for your work. when i use consistence loss on focs model with pascalvoc dataset, the accuracy always decrease, do you have any suggestion ?

**Comment by soo89** [comment]
What is the focs model?

I have checked it several times recently, and there is no problem with the SSD 300 + CSD model. (Sometimes, it works better than reported score)

However, there is a slight variation in the SSD 512 + CSD model. (75.4~75.8) 
(It is still much higher than the supervised learning score(73.3).)

**Comment by ZJUZQ** [comment]
> What is the focs model?
> 
> I have checked it several times recently, and there is no problem with the SSD 300 + CSD model. (Sometimes, it works better than reported score)
> 
> However, there is a slight variation in the SSD 512 + CSD model. (75.4~75.8)
> (It is still much higher than the supervised learning score(73.3).)

thank you for you reply.
fcos model refer to https://arxiv.org/abs/1904.01355

when i do experiment with fcos model, i finetune from supervised model. For each batch, i choose the topk (e.g., 20) detections to apply consistency loss (reg loss and cls loss). however, i found that, when apply cons_reg_loss, the normal reg loss seems to increase, is there anything i ignored ?

**Comment by soo89** [comment]
The regression method is different from SSD, so I think you should check it well.

For example, in loc consistency loss, the flipping transformation makes delta x be calculated in the opposite direction.

However, in fcos, you should check l and r simultaneously.

**Comment by ZJUZQ** [comment]
> The regression method is different from SSD, so I think you should check it well.
> 
> For example, in loc consistency loss, the flipping transformation makes delta x be calculated in the opposite direction.
> 
> However, in fcos, you should check l and r simultaneously.

yes, for reg, i already flipped the feature map and [l, t, r, b]
   ...
  flip_box_reg = flip_box_reg_mc[fpn_level].reshape(_n, _c, 4, _h, _w).flip((-1,)).permute(0, 1, 3, 4, 2)
  ...
  filter_flip_reg = torch.cat(filter_flip_reg, dim=0)[:, [2, 1, 0, 3]]
   
i find that, the csd_reg_loss (e.g., 0.001) is very samll, compared to reg_loss (e.g., 0.27)

**Comment by soo89** [comment]
I don't have any idea because I haven't seen that code.

I hope you check it well and have good results.

**Comment by ZJUZQ** [comment]
> I don't have any idea because I haven't seen that code.
> 
> I hope you check it well and have good results.

thank you anyway, i will check it again.
